Documentation/configuration.md

```diff
@@ -149,7 +149,110 @@ In `/etc/rkt/auth.d/specific-tectonic.json`:
 }
 ```
 
-The result is that when downloading data from `kubernetes.io` we still
-send `Authorization: Bearer common-token`, but when downloading from
-`coreos.com` - `Authorization: Basic Zm9vOmJhcg==`. And for
-`tectonic.com` - `Authorization: Bearer tectonic-token`.
+The result is that when downloading data from `kubernetes.io` `rkt`
+still sends `Authorization: Bearer common-token`, but when downloading
+from `coreos.com` - `Authorization: Basic Zm9vOmJhcg==` (`foo:bar`
+encoded in base64). And for `tectonic.com` - `Authorization: Bearer
+tectonic-token`.
+
+### rktKind: `dockerAuth`
+
+This kind of configuration is used to set up necessary credentials
+when downloading data from docker indices. The configuration files
+should be placed inside either `auth.d` or `docker.d` subdirectories
+(that is - in `/usr/lib/rkt/{auth.d,docker.d}` or in
+`/etc/rkt/{auth.d,docker.d}`).
+
+#### rktVersion: `v1`
+
+##### Description and examples
+
+This version of `dockerAuth` configuration specifies two additional
+fields: `indices` and `credentials`.
+
+The `indices` field is an array of strings describing docker indices
+for which following credentials should be used. A short list of
+popular docker indices is below. This field has to be specified and
+cannot be empty.
+
+`credentials` field holds the necessary data to authenticate against
+docker index. This field has to be specified and cannot be empty.
+
+Currently docker indices only support basic HTTP authentication, so
+`credentials` field has two subfields - `user` and `password`. These
+fields have to be specified and cannot be empty.
+
+Some popular docker indices:
+* index.docker.io (this is is used when no docker index is specified
+  in URL, like in `docker://redis`)
+* quay.io
+* gcr.io
+
+Example of dockerAuth config:
+```
+{
+	"rktKind": "dockerAuth",
+	"rktVersion": "v1",
+	"indices": ["index.docker.io", "quay.io"],
```

Comment:
> My impression was that index was the first "point of contact" in docker hub API - getting metadata and whatnot. Another impression I had was that the basic HTTP authentication is done against that index and later we just receive a docker token for authentication.

That's correct.

This is confusing. Docker Hub has `index.docker.io` and `registry.hub.docker.com`. According to [this](https://docs.docker.com/reference/api/hub_registry_spec/), something like quay.io or gcr would be a *private registry*. They don't mention anything about an index there and, in the rest of the documentation, when they talk about an index they refer only to the Docker Hub.

They don't expect people running their own indexes except for this *private registry* category and that's why a they describe a registry as the part that just serves images, without authentication. But things like quay.io or gcr are full registries (in the broad sense of the term) with authentication included so I think registry is the correct term here.

TL;DR

Use registry :stuck_out_tongue_winking_eye: